sympy/simplify/tests/test_sqrtdenest.py

```diff
@@ -183,11 +181,13 @@ def test_issue_5653():
         sqrt(2 + sqrt(2 + sqrt(2)))) == sqrt(2 + sqrt(2 + sqrt(2)))
 
 def test_issue_12420():
-    I = S.ImaginaryUnit
-    assert _sqrt_match(4 + I) == []
     assert sqrtdenest((3 - sqrt(2)*sqrt(4 + 3*I) + 3*I)/2) == I
```

Comment:
It seems that `Add._eval_power` can handle this:
```
>>> 3 - sqrt(2)*sqrt(4 + 3*I) + 3*I
2*I
```

Reply:
We have

```
>>> sqrtdenest(S("(3 - sqrt(2)*sqrt(4 + 3*I) + 3*I)/2", evaluate=False))
I
```

though I would need to check if it's not just evaluating it somehow. We probably need better tests here.